suricata/update/loghandler.py

```diff
@@ -62,18 +66,53 @@ def emit(self, record):
             level_prefix = self.YELLOW
             message_prefix = ""
 
-        self.stream.write("%s%s%s - <%s%s%s> -- %s%s%s\n" % (
-            self.GREEN,
-            self.formatTime(record),
-            self.RESET,
-            level_prefix,
-            record.levelname.title(),
-            self.RESET,
-            message_prefix,
-            self.mask_secrets(record.getMessage()),
-            self.RESET))
+        if os.isatty(self.stream.fileno()):
+            self.stream.write("%s%s%s - <%s%s%s> -- %s%s%s\n" % (
+                self.GREEN,
+                self.formatTime(record),
+                self.RESET,
+                level_prefix,
+                record.levelname.title(),
+                self.RESET,
+                message_prefix,
+                self.mask_secrets(record.getMessage()),
+                self.RESET))
+        else:
+            self.stream.write("%s - <%s> -- %s\n" % (
+                self.formatTime(record),
+                record.levelname.title(),
+                self.mask_secrets(record.getMessage())))
 
     def mask_secrets(self, msg):
         for secret in secrets:
             msg = msg.replace(secret, "<%s>" % secrets[secret])
         return msg
+
+
+class LessThanFilter(logging.Filter):
+    def __init__(self, exclusive_maximum, name=""):
+        super(LessThanFilter, self).__init__(name)
+        self.max_level = exclusive_maximum
+
+    def filter(self, record):
+        return 1 if record.levelno < self.max_level else 0
+
+
+def configure_logging():
+    if os.fstat(sys.stdout.fileno()) == os.fstat(sys.stderr.fileno()):
+        filter_stdout = True
+    else:
+        filter_stdout = False
+    logger = logging.getLogger()
+    logger.setLevel(logging.NOTSET)
+    logging_handler_out = SuriColourLogHandler(sys.stdout)
+    logging_handler_out.setLevel(logging.DEBUG)
+    if filter_stdout:
+        logging_handler_out.addFilter(LessThanFilter(logging.WARNING))
+        logger.addHandler(logging_handler_out)
+    if not filter_stdout:
```

Comment:
Do we need this checking for critical?  Critical is higher than error if you consider [debug,info,warn,error,critical] so it should be covered by the less than filter for warning.

Sorry if the other messages were unclear in this respect.  When I often say info, or warn, error what I really mean is [debug,info] and [warn,error,critical].  So I think we can get by with just filtering on logging.WARNING, then adding the handler outside of the conditional.  Otherwise, I think this is good.